packages/node/src/login/oidc/ClientRegistrar.spec.ts

```diff
@@ -22,19 +22,43 @@
 import { randomUUID } from "crypto";
 import { jest, it, describe, expect } from "@jest/globals";
 import { mockStorageUtility } from "@inrupt/solid-client-authn-core";
+import type * as OpenIdClient from "openid-client";
+import type { IOpenIdDynamicClient } from "core";
 import ClientRegistrar from "./ClientRegistrar";
 import {
   IssuerConfigFetcherFetchConfigResponse,
-  mockDefaultIssuerMetadata,
   mockIssuerMetadata,
 } from "./__mocks__/IssuerConfigFetcher";
-import {
-  mockClientConfig,
-  mockDefaultClientConfig,
-} from "./__mocks__/ClientRegistrar";
+import { mockDefaultClientConfig } from "./__mocks__/ClientRegistrar";
 
 jest.mock("openid-client");
 
+const mockClientRegistration = (
+  issuerMetadata: Record<string, string | undefined>,
+  clientMetadata: OpenIdClient.ClientMetadata,
+) => {
+  // Sets up the mock-up for DCR
+  const { Issuer } = jest.requireMock("openid-client") as jest.Mocked<
+    typeof OpenIdClient
+  >;
+  const mockedIssuerConfig = mockIssuerMetadata(issuerMetadata);
+  const mockedIssuer: OpenIdClient.Issuer<OpenIdClient.BaseClient> = {
+    metadata: mockedIssuerConfig,
+    Client: {
+      register: (jest.fn() as any).mockResolvedValueOnce({
+        metadata: clientMetadata,
+      }),
+      // The assertions are required because we only mock we is strictly necessary for our tests.
```

Comment:
```suggestion
      // The assertions are required because we only mock what is strictly necessary for our tests.
```
Is this what you mean?